dev/NumberBox/NumberBoxParser.cpp

```diff
@@ -0,0 +1,301 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "NumberBoxParser.h"
+#include "Utils.h"
+
+static constexpr wstring_view c_numberBoxOperators{ L"+-*/^"sv };
+
+// Returns list of MathTokens from expression input string. If there are any parsing errors, it returns an empty vector.
+std::vector<MathToken> NumberBoxParser::GetTokens(const wchar_t* input, winrt::INumberParser numberParser)
+{
+    auto tokens = std::vector<MathToken>();
+
+    size_t index = 0;
+    bool error = false;
+    bool expectNumber = true;
+    while (input[0] != '\0' && !error)
+    {
+        // Skip spaces
+        auto nextChar = input[0];
+        if (nextChar != L' ')
+        {
+            if (expectNumber)
+            {
+                if (nextChar == '(')
+                {
+                    // Open parens are also acceptable, but don't change the next expected token type.
+                    tokens.push_back(MathToken(MathTokenType::Parenthesis, nextChar));
+                }
+                else
+                {
+                    double value = NAN;
+                    const auto charLength = GetNextNumber(input, numberParser, value);
+
+                    if (charLength > 0)
+                    {
+                        tokens.push_back(MathToken(MathTokenType::Numeric, value));
+                        input += charLength - 1; // advance the end of the token
+                        expectNumber = false; // next token should be an operator
+                    }
+                    else
+                    {
+                        // Error case -- next token is not a number
+                        error = true;
+                    }
+                }
+            }
+            else
+            {
+                if (c_numberBoxOperators.find(nextChar) != std::wstring::npos)
```

Comment:
>c_numberBoxOperators.find(nextChar) != std::wstring::npos [](start = 20, length = 57)

So, 3(12/15) will fail as well? there is an implicit * in that calculation

Reply:
Yes, it doesn't support implicit operators currently.